components/google_drive/actions/create-file-from-template/create-file-from-template.mjs

```diff
@@ -0,0 +1,109 @@
+import googleDrive from "../../google_drive.app.mjs";
+import Mustaches from "google-docs-mustaches";
+
+const MODE_GOOGLE_DOC = 0;
+const MODE_PDF = 1;
+const MODE_GOOGLE_DOC_PDF = 2;
+
+export default {
+  key: "google_drive-create-file-from-template",
+  name: "Create New File From Template",
+  description: "Create a new google doc file from template",
+  version: "0.0.1",
+  type: "action",
+  props: {
+    googleDrive,
+    templateId: {
+      propDefinition: [
+        googleDrive,
+        "fileId",
+      ],
+      description:
+        "Id of the file you want to use as a template.",
+    },
+    folderId: {
+      propDefinition: [
+        googleDrive,
+        "folderId",
+      ],
+      description:
+        "Folder id of the newly created google doc and pdf.",
+    },
+    name: {
+      propDefinition: [
+        googleDrive,
+        "fileName",
+      ],
+      description:
+        "Name of the file you want to create (eg. `myFile` will create a google doc called `myFile` and a pdf called `myFile.pdf`)",
+    },
+    mode: {
+      type: "string",
+      label: "Mode",
+      description: "Select if you want to create the google doc, the pdf or both files.",
+      async options() {
+        return [
+          {
+            label: "Google doc",
+            value: 0,
+          },
+          {
+            label: "Pdf",
+            value: 1,
+          },
+          {
+            label: "Google doc & Pdf",
+            value: 2,
+          },
+        ];
+      },
+    },
```

Comment:
I would make this prop an array of strings so the user can check one or both options.
```suggestion
    mode: {
      type: "string[]",
      label: "Mode",
      description: "Select if you want to create the google doc, the pdf or both files.",
      options: [
          "Google Doc",
          "PDF",
      ],
    },
```

Reply:
No because if I want to create a Google Doc and the pdf file, it is not possible.
What I must do is replace this:
```javascript
        return [
          {
            label: "Google doc",
            value: 0,
          },
          {
            label: "Pdf",
            value: 1,
          },
          {
            label: "Google doc & Pdf",
            value: 2,
          },
        ];
```

By this:
```javascript
        return [
          {
            label: "Google doc",
            value: MODE_GOOGLE_DOC,
          },
          {
            label: "Pdf",
            value: MODE_PDF,
          },
          {
            label: "Google doc & Pdf",
            value: MODE_GOOGLE_DOC_PDF,
          },
        ];
```